### PR TITLE
IEMEREDU13-60(mediator): Implement destination service

### PR DESCRIPTION
**Pull request by myfosse**

### CODE REVIEW

File: src/main/java/com/leverx/mediator/controller/MultiController.java [hunk 1]
```diff
@@ -36,7 +36,7 @@ public MultiController(final MultiService multiService, final Auth auth) {
 
   @GetMapping
   public ResponseEntity<UserCatDogListResponse> getAllLists(
```
Comment (ilyaaleks) [inline src/main/java/com/leverx/mediator/controller/MultiController.java:38]: `getAllLists` - > `getAllPets`

File: src/main/java/com/leverx/mediator/repository/destination/impl/CatDestinationRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  @Value("${sap.link.cats}")
+  private final String catsLink;
+
+  @Override
+  public CatResponse save(final CatRequest cat) {
+    log.info("CatRepository. Save cat: {}", cat);
+
+    CatResponse catResponse;
+
+    HttpPost post = new HttpPost(catsLink);
```
Comment (ilyaaleks) [inline src/main/java/com/leverx/mediator/repository/destination/impl/CatDestinationRepositoryImpl.java]: Inappropriate use of the destination service. You can make request by this way:
```
final Destination destination = getDestination(destinationName);
        HttpClient client = HttpClientAccessor.getHttpClient(destination.asHttp());
        HttpGet httpGetRequest = new HttpGet("/cats");
        HttpResponse response = client.execute(httpGetRequest);

```
and then handle a response...
This service was just created to set the relative paths.

File: src/main/java/com/leverx/mediator/repository/destination/impl/CatDestinationRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    try {
+      post.setEntity(new StringEntity(mapper.writeValueAsString(cat)));
+      post.setHeader("Accept", "application/json");
+      post.setHeader("Content-type", "application/json");
```
Comment (ilyaaleks) [inline src/main/java/com/leverx/mediator/repository/destination/impl/CatDestinationRepositoryImpl.java]: To many work in service methods. Try to decouple logic of configuration http client and business logic.

File: src/main/java/com/leverx/mediator/repository/resttemplate/impl/UserRestTemplateRepositoryImpl.java [hunk 4]
```diff
@@ -28,9 +28,9 @@
...
 @Slf4j
 @RequiredArgsConstructor
-public class UserRepositoryImpl implements UserRepository {
+public class UserRestTemplateRepositoryImpl implements UserRepository {
 
-  @Value("${leverx.com.link.sap.users}")
+  @Value("${sap.link.users}")
   private final String usersLink;
 
   private final Auth auth;
```
Comment (ilyaaleks) [inline src/main/java/com/leverx/mediator/repository/resttemplate/impl/UserRestTemplateRepositoryImpl.java:35]: You can register RestTemplate as spring bean.
More info: https://stackoverflow.com/questions/22989500/is-resttemplate-thread-safe

File: src/test/java/com/leverx/mediator/MediatorApplicationTests.java [hunk 5]
```diff
@@ -3,11 +3,10 @@
...
 @SpringBootTest
-class MediatorApplicationTests {
+public class MediatorApplicationTests {
 
   @Test
-  void contextLoads() {
-  }
-
+  void contextLoads() {}
 }
```
Comment (ilyaaleks) [inline src/test/java/com/leverx/mediator/MediatorApplicationTests.java:12]: Where are the tests?:)

Comment (ilyaaleks) [reply src/test/java/com/leverx/mediator/MediatorApplicationTests.java:12]: And one more comment:
Is `userId` varaible in `UserTransactionImpl` thread-safe? And what happend, if transaction not finished yet, and we should process a new request from client?

File: src/main/java/com/leverx/mediator/controller/MultiController.java (hunk 1 above)
Comment (myfosse) [reply src/main/java/com/leverx/mediator/controller/MultiController.java:38]: > `getAllLists` - > `getAllPets`

This function return all lists of entities(list of users, list of cats, list of dogs). So I don't see reason to rename this function

**Review by Uladzislau-Kukoba — APPROVED** [review]

**Pull request merged**